# Instructions

This assessment concerns the well-known tips dataset and the Python packages seaborn and jupyter. The project is broken into three parts, as follows:

1. **Description**: Create a git repository and make it available online for the lecturer to clone. The repository should contain all your work for this assessment. Within the repository, create a jupyter notebook that uses descriptive statistics and plots to describe the tips dataset.

2. **Regression**: To the above jupyter notebook add a section that discusses and analyses whether there is a relationship between the total bill and tip amount.

3. **Analyse**: Again using the same notebook, analyse the relationship between the variables within the dataset. You are free to interpret this as you wish — for example, you may analyse all pairs of variables, or select a subset and analyse those.

In [1]:
# Show plots in notebook
%matplotlib inline

In [2]:
# Import packages.
import seaborn as sns
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Part 1. Description

In this assessment, I work with the tips dataset which is inbuild in a seaborn package. The data is this dataset were recorded in a restaurant in US by a food server on all customers they served during an interval of two and a half months in early 1990. The restaurant, located in a suburban shopping mall, was part of a national chain and offered to seat in a non-smoking section to patrons who requested it.

First, let's load the tips dataset and see the summary statistic of the dataset.

In [3]:
# Load the tips dataset which is part of the seaborn package.
tips = sns.load_dataset("tips")

In [4]:
# Show the tips dataset.
tips

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

We can see that tips dataset has 244 rows and 7 columns:
* total_bill- cost of the meal including tax, in US dollars
* tip- gratuity in US dollars
* sex- sex of person paying for the meal (male or female)
* smoker- smoker in party (no or yes)
* day- day of the week (Thursday, Friday, Saturday, Sunday)
* time- night or day (dinner or lunch)
* size- size of the party (from 1 to 6)

In [5]:
# Returns summary statistics of the dataset
tips.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

We have 244 values in each numerical cell, no values are missing. 
* The minimul value of total_bill is 3.07, the average value is 19.7859 and the maximum value is 50.81. The standard deviation is 8.9024 so the values are not close to the mean but are spread out over a wider range.  
* The minimul value of tip is 1, the average value is 2.9983 and the maximum value is 10. The standard deviation is 1.3836 so the values are close to mean.
* The minimul value of size is 1, the average value is 2.5697 and the maximum value is 6. The standard deviation is 0.9511 so the values are close to mean.

# Part 2. Regression

# Part 3. Analyse

# References


[Seaborn- An Introduction to seaborn](https://seaborn.pydata.org/introduction.html)

[Towards Data Science- Analyze the data through data visualization using Seaborn](https://towardsdatascience.com/analyze-the-data-through-data-visualization-using-seaborn-255e1cd3948e)

[Kaggle- Seaborn tips dataset](https://www.kaggle.com/ranjeetjain3/seaborn-tips-dataset)